openpgp/write.go

```diff
@@ -499,14 +515,27 @@ func (c noOpCloser) Close() error {
 	return nil
 }
 
-func handleCompression(compressed io.WriteCloser, config *packet.Config) (data io.WriteCloser, err error) {
+func handleCompression(compressed io.WriteCloser, candidateCompression []uint8, config *packet.Config) (data io.WriteCloser, err error) {
 	data = compressed
-	if algo := config.Compression(); algo != packet.CompressionNone {
+	confAlgo := config.Compression()
+	if confAlgo == packet.CompressionNone {
+		return
+	}
+	var finalAlgo packet.CompressionAlgo
```

Comment:
In this case we're using a default value to then check it later on line 533.
```suggestion
	var finalAlgo = packet.CompressionNone
```

Reply:
it is actually set here as default value due to it being zero. I could however do it explicitly.